test/fixtures/nested_minitest_tests.rb

```diff
@@ -0,0 +1,21 @@
+class ParentTest < Minitest::Test
+  def test_public; end
+
+  private
+
+  class FirstChildTest < Minitest::Test
```

Comment:
From my experience, such confusing structure usually comes from legacy projects. For example, IRB [used to have nested test classes](https://github.com/ruby/irb/blob/v1.4.3/test/irb/test_cmd.rb) before I refactored it. Although it didn't switch visibility that much, it's not very far from the test case I described here. I think it's in such confusing cases that our tool will provide even more value to users. 

Of course, there's always a balance there. But given this support is not too complicated, I think we should do that.

Reply:
Looking at [ruby style guide for minitest](https://minitest.rubystyle.guide/), I don't see anything oppose to the above cases. So IMO not supporting them because "we feel it's a bad pattern" isn't very convincing.

Reply:
I think there's a misunderstanding of the principles of the Ruby LSP. We're intended to be an opinionated LSP.

Our priorities are **always** performance and stability over features, so we will never make the decision to cover more niche cases at the expense of those two. Increasing complexity of the codebase to cover every possible scenario is also something we will not do. There's no need to convince anyone, that's the goal of the project and it's what we will do.

In this particular case, the increase in complexity is not very high yet, but we will need to properly support `--testcase` (or else it doesn't work properly) and we need to make sure the test controller displays the nesting correctly.

If this ends up being straightforward, then fine. But if it grows in complexity too much, then let's only support multiple test classes in the same file, but not nested test classes.

Reply:
The lack of `--testcase` support is also something I want to fix in a follow up PR because it means 1 file can only have 1 test class. Otherwise the test class' code lends doesn't work properly. 

For example, if you click code lens on any test class in this [IRB test file](https://github.com/ruby/irb/blob/master/test/irb/test_cmd.rb), it always runs all test cases in that file. It also doesn't work in
- some `ruby/debug`'s test files ([example](https://github.com/ruby/debug/blob/master/test/console/break_test.rb))
- The [`ruby/pp` gem](https://github.com/ruby/pp/blob/master/test/test_pp.rb)

I probably didn't do a good job listing all problems I've seen so far, but I feel frustrated being implied not understanding the principle of the project while I'm just trying to make Ruby LSP working properly in some Ruby Core projects.